File: moby-dick-vanilla-js/read-file.js
const {promises: fsPromises} = require('fs');

async function readFile() {
  try {
    const contents = await fsPromises.readFile('./text-files/moby-dick.txt', 'utf-8');
    const stopWordFile = await fsPromises.readFile('./text-files/stop-words.txt', 'utf-8');

    const lines = contents.split(/\r?\n/);
    const stopWords = stopWordFile.split(/\r?\n/);

    var counted = {}

    for(let line of lines) {
      let wordsInLine = line.split(' ').map(l => l.replace(/[^a-zA-Z0-9 ]/g, '').toLowerCase())
      for(let w of wordsInLine) {
        if(w && w !== '' && !stopWords.includes(w)) {
          if(counted[w]) {
            counted[w]++
          }else{
            counted[w] = 1
          }
        }
      }
    }

    var sorted = []

    for(let c in counted) {
      sorted.push({word: c, count: counted[c]})
    }

    sorted.sort((a,b) => {
      return b.count - a.count
    })

    return sorted.slice(0,100)

  } catch (err) {
    console.error(err);
  }

}

readFile()
  .then(res => console.log(res))
  .catch(err => err)

File: moby-dick-python/read-file.py
import re
def count():
  stopwords = []
  with open('./moby-dick-python/text-files/stop-words.txt') as swf:
    stop_word_file = swf.readlines()

  for line in stop_word_file:
    stopwords.append(line.strip())

  counted = {}
  sortedWords = {}
  with open('./moby-dick-python/text-files/moby-dick.txt', errors='ignore') as mbf:
    moby_dick = mbf.readlines()
  for line in moby_dick:
    wordsInLine = line.split()
    for w in wordsInLine:
      word = re.sub(r"[^a-zA-Z0-9]","",w.lower())
      if word not in stopwords:
        if word not in counted:
          counted[word] = 1
        else:
          counted[word] = counted[word] + 1

  sort_words = sorted(counted.items(), key=lambda x: x[1], reverse=True)

  iterator = 0
  for i in sort_words:
    if iterator > 99:
      break
    sortedWords[i[0]] = i[1]
    iterator += 1
  
  return sortedWords

def main():
  words = count()
  topCount = list(words.values())[0]
  for key, val in words.items():
    perc = round((val/topCount)*50)
    meter = '|'
    i = 0
    while i <= 50:
      if i < perc:
        meter += '-'
      elif i == perc:
        meter += '>'
      else:
        meter += ' '
      i += 1
    meter += '| ' + str(val)
    print(key)
    print(meter)

if __name__ == '__main__':
  main()